spyder_kernels/customize/spydercustomize.py

```diff
@@ -727,6 +727,7 @@ def set_spyder_pythonpath():
     pypath = os.environ.get('SPY_PYTHONPATH')
     if pypath:
         pathlist = pypath.split(os.pathsep)
-        sys.path.extend(pathlist)
+        # User provided paths should take priority
+        [sys.path.insert(0, p) for p in reversed(pathlist)]
```

Comment:
Why not `sys.path = pathlist + sys.path`?

Reply:
Actually it probably doesn't matter, the number of paths is not that large

Reply:
`sys.path` should not be reassigned in that manner because `sys.path` is a mutable object that may have been imported somewhere else prior to being manipulated here. Any namespace that has already imported `sys.path` will continue to point to the original memory address location, while subsequent imports will point to the new `sys.path`, thus `sys.path` should be manipulated "in place". Both `.extend()` and `.insert()` methods do this.